### DISCUSSION TITLE
Proxy config is ignored by Windows-based docker image

### DISCUSSION THREAD

**Discussion by boldt** [dotnet/dotnet-docker#4121] [Q&A]
### Describe the Bug

When using the Windows-based docker image for dotnet sdk, dotnet restore takes a huge amount of time or simply fails and can't overcome the corporate proxy. We set the proxy configuration as

* part of the nuget.config file (as described [in the documentation](https://docs.microsoft.com/en-us/nuget/reference/nuget-config-file) under the config section),
* environment variables via Gitlab CI / Gitlab Runner (`http_proxy`, `HTTP_PROXY`, `https_proxy`, `HTTPS_PROXY`, `no_proxy`, `NO_PROXY`) and
* build args as ENV in a derived docker image.

This happens on a Win 10 Professional (20H2) VM used as build-server with Gitlab CI and local Win 10 Professional (20H2) developer machines. Running `dotnet restore` outside a container works as expected. 

The linux docker image works with the same setup. There the environment variables will also be injected by the Gitlab Runner and used in the linux dotnet sdk image for restoring the dependencies of projects.

### Steps to Reproduce

Setup an environment with a Win 10 Professional (20H2), a proxy and run the [mcr.microsoft.com/dotnet/sdk:6.0](http://mcr.microsoft.com/dotnet/sdk:6.0) docker image with a small project like the default ASP.NET api template. Use one of the above mentioned ways to inject the proxy settings into the dotnet sdk within the docker image and run a dotnet restore with the project copied to the docker image.

#### Expected Behaviour

A dotnet restore within the docker image uses the `nuget.config` with set proxy settings or environment variables and thus works through a corporate proxy as it does on the host system. 

#### Actual Behaviour

A `dotnet restore` fails with "error : Unable to load the service index for source https://api.nuget.org/v3/index.json." on a local machine or takes a huge amount of time in the VM.

The following picture is from our build server:

![grafik](https://user-images.githubusercontent.com/750793/193211693-85684dc4-e825-4e45-9f7f-7088af9db574.png)

The same build takes around 1 minute on the VM without docker or on a different Linux-based system with the docker image compared to the Windows-based docker job, which takes 20 minutes.

Our administrator looked at the logs and saw that the VM tries to bypass the proxy until it finally settles with it.

### Other Information

* OS: Win 10 Professional (20H2)
* CPU Arch: x64
* SDK Version: 6.0.400

### Output of `docker version`

![grafik](https://user-images.githubusercontent.com/750793/193213259-e39f5c91-7309-42d5-bc3d-5ab8f26e471c.png)

### Output of `docker info`

![grafik](https://user-images.githubusercontent.com/750793/193214513-00ddcd33-c8fe-42a1-9538-f44e75dad403.png)

**Comment by ghost** [comment]
I couldn't figure out the best area label to add to this issue. If you have write-permissions please help me learn by adding exactly one area label.

**Comment by mthalman** [comment]
> Our administrator looked at the logs and saw that the VM tries to bypass the proxy until it finally settles with it.

Are you able to provide a portion of these logs, sanitized if necessary?

**Reply by boldt** [reply]
I will provide them, as soon I get them.

**Comment by mthalman** [comment]
Is this specific to NuGet? Are you able to make a simple ping or other web request from the Windows container?

**Reply by boldt** [reply]
I can confirm, that ping works from the Windows container. As you can see, a restore of a small MWE takes 11 minutes:

![grafik](https://user-images.githubusercontent.com/750793/198643818-4d7cd3c3-dbb7-4ef5-ae96-e900f70f5ee0.png)
